### PR TITLE
QFJ-906 Source code language level set to JDK 7.

### PR DESCRIPTION
**Pull request by guidomedina**
Source code language level set JDK 7, can be set to JDK 8 later but first we need to keep it at JDK 7 for easy support of both 1.7.x and 1.6.x

### CODE REVIEW

**Comment by guidomedina** [comment]
I will give you another branch sometime next week or later with JDK 8 optimizations, like using `computeIfAbsent/Present ` for Maps with a loading function, it is much better for caching and stuff with new JDK 8 concurrent changes.

**Comment by guidomedina** [comment]
Added a dom4j change to new version 2.0.0 of this year, under new Maven group:
http://search.maven.org/#artifactdetails%7Corg.dom4j%7Cdom4j%7C2.0.0%7Cjar

File: quickfixj-core/src/main/java/quickfix/Initiator.java [hunk 1]
```diff
@@ -30,42 +30,42 @@
      *
      * @see quickfix.SessionFactory#SETTING_CONNECTION_TYPE
      */
-    public static final String SETTING_RECONNECT_INTERVAL = "ReconnectInterval";
+    String SETTING_RECONNECT_INTERVAL = "ReconnectInterval";
```
Comment (chrjohn) [inline quickfixj-core/src/main/java/quickfix/Initiator.java]: Is there any specific reason to make this package-private? I have also seen it for other constants in this PR.

**Comment by guidomedina** [comment]
IntelliJ says that modifiers at interfaces are redundant, in an interface they are always `static final` for constants.

File: quickfixj-core/src/main/java/quickfix/Initiator.java (hunk 1 above)
Comment (guidomedina) [inline quickfixj-core/src/main/java/quickfix/Initiator.java]: IntelliJ says that modifiers at interfaces are redundant, in an interface they are always static final for constants.

File: quickfixj-core/src/main/java/quickfix/Session.java [hunk 2]
```diff
@@ -344,7 +343,7 @@
 
     public static final String SETTING_MAX_SCHEDULED_WRITE_REQUESTS = "MaxScheduledWriteRequests";
```
Comment (chrjohn) [inline quickfixj-core/src/main/java/quickfix/Session.java]: These constants remain as public...

Comment (guidomedina) [inline quickfixj-core/src/main/java/quickfix/Session.java]: It is only at interfaces.

File: quickfixj-core/src/main/java/quickfix/Initiator.java (hunk 1 above)
Comment (chrjohn) [inline quickfixj-core/src/main/java/quickfix/Initiator.java]: Of course. Sorry, I did not realize it was an interface. :)

Comment (guidomedina) [inline quickfixj-core/src/main/java/quickfix/Initiator.java]: Well, IntelliJ inspection, but that's a Java rule TBH.

File: quickfixj-core/src/main/java/quickfix/Session.java (hunk 2 above)
Comment (chrjohn) [inline quickfixj-core/src/main/java/quickfix/Session.java]: Just deleted the comment ;)

**Comment by chrjohn** [comment]
Looks good to me. Thank you.
Will cherry-pick it later.

**Comment by guidomedina** [comment]
Good, next week I will take another look from a Java 8 perspective.

**Comment by chrjohn** [comment]
Thanks, have a nice weekend.

**Comment by chrjohn** [comment]
Cherry picked with commit 629b801446431f06e0426aebcf669a2eca341a69

**Pull request closed without merging**